test/hotspot/jtreg/compiler/jvmci/compilerToVM/LookupNameAndTypeRefIndexInPoolTest.java

```diff
@@ -95,19 +96,28 @@ private static void validate(ConstantPool constantPoolCTVM,
         if (entry == null) {
             return;
         }
-        int index = cpi;
-        String cached = "";
-        int cpci = dummyClass.getCPCacheIndex(cpi);
-        if (cpci != ConstantPoolTestsHelper.NO_CP_CACHE_PRESENT) {
-            index = cpci;
-            cached = "cached ";
+        int opcode;
+        int index = dummyClass.getCPCacheIndex(cpi);
+        Asserts.assertTrue(index != ConstantPoolTestsHelper.NO_CP_CACHE_PRESENT, "the class must have been rewritten");
+        // Select an arbitrary bytecode of the type associated with the Constant pool entry
+        switch(cpType) {
+          case CONSTANT_FIELDREF:
+            opcode = Bytecodes.GETFIELD;
+            break;
+          case CONSTANT_METHODREF:
+          case CONSTANT_INTERFACEMETHODREF:
+            opcode = Bytecodes.INVOKEVIRTUAL;
+            break;
+          case CONSTANT_INVOKEDYNAMIC:
+            opcode = Bytecodes.INVOKEDYNAMIC;
+            break;
+          default:
+            throw new Error("Unexpected consant pool entry");
```

Comment:
Typo: consant

Reply:
I wonder if this should be refactored into a little function toOpcodeType() or something like that.  It probably should be for these tests.